docs/API-DESIGN.md

```diff
@@ -0,0 +1,865 @@
+   </td>
+   <td>Already present in the current API. 
+   </td>
+  </tr>
+  <tr>
+   <td><code>GET /cluster/{cluster_name}/tables</code>
+   </td>
+   <td><code>ClusterSchema(ctx context.Context, name string) (map[string]*Keyspace, error)</code>
+   </td>
+   <td><code>Schema</code> rather than <code>Tables</code> since it is a map of tables keyed by keyspace.
+   </td>
+  </tr>
+  <tr>
+   <td><code>POST /cluster</code>
+<p>
+<code>POST /cluster/auth</code>
+   </td>
+   <td><code>CreateCluster(ctx context.Context, seed string, jmxOptions *ClusterJmxOptions) (string, error)</code>
+   </td>
+   <td>Already present in the current API. Will use the special<code> /auth</code> endpoint if authenticated, otherwise the regular one.
+   </td>
+  </tr>
+  <tr>
+   <td><code>PUT /cluster/{cluster_name}</code>
+<p>
+<code>PUT /cluster/{cluster_name}/auth</code>
+   </td>
+   <td><code>UpdateCluster(ctx context.Context, name string, newSeed string, jmxOptions *ClusterJmxOptions) error</code>
+   </td>
+   <td>Already present in the current API. Will use the special<code> /auth</code> endpoint if authenticated, otherwise the regular one.
+   </td>
+  </tr>
+  <tr>
+   <td><code>DELETE /cluster/{cluster_name}</code>
+   </td>
+   <td><code>DeleteCluster(ctx context.Context, name string, force bool) error</code>
+   </td>
+   <td>Already present in the current API. 
+   </td>
+  </tr>
+</table>
+
+
+### Types
+
+Note: the `NodeState` struct attempts to capture datacenter and rack names in a user-friendly manner, but does not correspond 1:1 with the JSON payload and will require a custom JSON unmarshaller. The current API already does this translation.
+
+```
+type Cluster struct {
+  Name            string
+  JmxUsername     string
+  JmxPasswordSet  bool
+  Seeds           []string
+  RepairRuns      []*RepairRun
+  RepairSchedules []*RepairSchedule
+  NodeStates      []*NodeState
+}
+
+type Keyspace struct {
+  Name   string
+  Tables map[string]*Table
+}
+
+type Table struct {
+  Name string
+}
+
+type NodeState struct {
+  SourceNode  string
+  TotalLoad   float64
+  Endpoints   []string
+  Datacenters map[string]*Datacenter
+}
+
+type Datacenter struct {
+  Name  string
+  Racks map[string]*Rack
+}
+
+type Rack struct {
+  Name      string
+  Endpoints []*Endpoint
+}
+
+type Endpoint struct {
+  Endpoint       string  
+  Datacenter     string  
+  Rack           string  
+  HostId         string  
+  Status         string  
+  Severity       float64 
+  ReleaseVersion string  
+  Tokens         string  
+  Load           float64 
+}
+```
+
+
+
+## RepairRun Resource
+
+
+### Methods
+
+
+<table>
+  <tr>
+   <td><strong>REST endpoint</strong>
+   </td>
+   <td><strong>Client API method</strong>
+   </td>
+   <td><strong>Comments</strong>
+   </td>
+  </tr>
+  <tr>
+   <td><code>GET /repair_run</code>
+   </td>
+   <td><code>RepairRuns(ctx context.Context, searchOptions *RepairRunSearchOptions) ([]*RepairRun, error)</code>
```

Comment:
I'm thinking that methods returning lists of entities could in Go return maps instead, keyed by the entity id. This would allow for efficient location of a given entity in the list. Thoughts?

Reply:
Went ahead and implemented the idea in #6: looks nice in Go imho.

Reply:
I like the idea of using keyed entity identifiers.  Makes it easier to manage for the consumer.